### ISSUE TITLE
TH: Implement improved parser with EGAT 1-min data

### ISSUE THREAD

**Issue by shuuji3** [electricitymaps/electricitymaps-contrib#4401]
The current parser for Thailand only uses the single-point data from the website: https://energy.go.th/index.html. This value is from the latest generation value of the EGAT website: https://www.sothailand.com/sysgen/egat/.

This EGAT website actually has 1-minute granular data on both planning and real-time generation. The chart is drawn on `<canvas>` element but there is API endpoints for the actual data. It seems to return past data as well.

I'm going to implement an improved parser with these endpoints.

### actual generation
- example command
  ```
  curl 'https://www.sothailand.com/sysgen/ws/sysgen/actual' --data 'name=SYSTEM_GEN(MW)&day=01-01-2020'
  ```
- maybe since 2020-01-01 ?

### planned generation
- example command
  ```
  curl 'https://www.sothailand.com/sysgen/ws/sysgen/plan' --data 'day=01-04-2005'
  ```
- maybe since 2005-04-01 ?

## Screenshots

**EGAT website** (https://www.sothailand.com/sysgen/egat/)
<img width="1830" alt="screenshot of EGAT website" src="https://user-images.githubusercontent.com/1425259/181793618-0528082f-ec3a-42b6-82a1-e402aa4e890f.png">

**Translation of the panel**
![Translation of the panel](https://user-images.githubusercontent.com/1425259/181793727-79725a4e-ca1f-498b-af65-b010e58d17e5.png)

**Comment by PPsyrius** [comment]
I'm pretty sure EGAT's data only dated back to May 2nd, 2019:
![image](https://user-images.githubusercontent.com/19505219/182044563-1473cb6a-c8fc-4362-82ef-49c1a48249f0.png)

**Comment by Kongkille** [comment]
Can you elaborate @PPsyrius?

I'm running the above CURL command which is able to retrieve data for more recent datetimes. 

Eg:

`curl 'https://www.sothailand.com/sysgen/ws/sysgen/actual' --data 'name=SYSTEM_GEN(MW)&day=01-01-2022'`

Not entire sure how to interpret the data though

{"id":"1800142","name":"SYSTEM_GEN(MW)","day":"01-01-2022","timeStart":0,"lastClearTime":0,"list":[[0,14559.1],[60,14590.9],[120,14583.7] , ... }

**Comment by shuuji3** [comment]
Hi, @PPsyrius. Thank you for the additional information. I'm not familiar with Thailand's grid system so any information is helpful to understand it.

About the above data, maybe their available data varies depending on data type and endpoint URL. I tried to access http://control.egat.co.th but it seems down now. Is the URL still publicly accessible?

@Kongkille, the returned data is a bit cryptic but I think it can be interpreted as seconds from 00:00 in the specified day and generated electricity (MW). I wrote some comments about it in the code in #4407.
